core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp

```diff
@@ -237,6 +237,25 @@ inline size_t get_shmem_per_sm_prefer_l1(cudaDeviceProp const& properties) {
     return 0;
   }() * 1024;
 }
+
+// Assuming !cudaFuncSetCacheConfig(MyKernel, cudaFuncCachePreferL1) this
+// routine lists the approximate size for 50% cache size based on Cuda occupance
+// calculator spreadsheet.
+inline size_t get_shmem_per_sm_prefer_equal(cudaDeviceProp const& properties) {
+  int const compute_capability = properties.major * 10 + properties.minor;
+  return [compute_capability]() {
+    switch (compute_capability) {
+      case 70:
+      case 80: return 64;
```

Comment:
These number I agree with.  I need to think a bit more about the others.

Reply:
why return 0 for pre 70?

Reply:
Is it possible to set different cache config preferences for pre 70 architectures? According the the cuda occupancy calculator spreadsheet, they have only one cache size.

Reply:
3.x is actually configurable.

Reply:
Kepler has configurable shared memory size

Reply:
@rgayatri23 How exactly did you come up with these numbers? Can you add instructions to reproduce?

Reply:
@rgayatri23 in lieu of this routine, you can query the device property `sharedMemPerMultiprocessor` to get the maximum addressable shared memory. I can't seem to find a way to programmatically query if a device has a unified cache (== assignable carveout), but if it doesn't the various routines to set the cache carveout should be no-ops anyway.

Reply:
@masterleinad Sorry for the delayed response. I am now using the `sharedMemPerMultiprocessor` as suggested by @weinbe2 to determine the maximum allocatable shared memory. When the requested scratch memory is a quarter of it, I am asking the carveout of shared memory to be 25%, 50% if the requested scratch size is half of the allocatable amount and 75% otherwise.